### ISSUE TITLE
Service design for locking or deactivating a user

### ISSUE THREAD

**Issue by srtalbot** [cds-snc/platform-forms-client#2045]
### User story
As a form builder
I need to know when my account has been locked or deactivated
So that I can get in contact with CDS if this is in error

As a Service owner
I need to know when an employee of mine's account has been locked or deactivated
So that I can get in contact with Cds if this is in error

### Design considerations
- [x] What are the reasons we would lock an account?
- [x] What are the reasons we would deactivate an account?
- [x] How do we communicate warnings to a service owner or form builder
- [x] How do we transfer forms on an account to a service owner?
- [ ] What documentation do we need to write for the support team or ours to lock or deactivate an account?
- [x] What actions can be taken automatically?
- [x] 

### Handoff checklist

- [x] **Leading designer’s name:** 
- Carolyn  Connors

- [x] **Updated IA:**    
- https://miro.com/app/board/uXjVP1PAp5M=/?share_link_id=790795874038  

- [x] **Current state & future state design :**   
- https://www.figma.com/file/89fRce4gmy64UIArSZoAM8/v3.-GC-Forms-(builder-and-vault)?type=design&node-id=1256%3A53263&t=3jyO2neSOoGkwMU8-1

- [x] **System & Error messaging** 

- [x] **Bilingual content :** 

- [x] I**Interaction flow example & annotation:** 
- https://www.figma.com/file/89fRce4gmy64UIArSZoAM8/v3.-GC-Forms-(builder-and-vault)?type=design&node-id=1427%3A52485&t=6Vnj61J3qy1yRkc9-1

- [x] **Semantic layout & accessibility annotation**
- Document title
- Document regions
- H1 (hidden or visible?)
- Text Tags
- List Tags
- Form controls
- Aria roles, states and properties  (When no semantic tag is available, include aria roles )
- Alternative text  (All images must have bilingual alternative text)
- Tab order (For any complex interactions, include tab order )

- [x] **Tailwind classes**
- font sizes
- colours
- hover & focus states

**Comment by samsadasivan** [comment]
Flowchart before screens

**Comment by bryan-robitaille** [comment]
A few tidbits to keep in mind:
- User accounts will automatically unlock after 15 min.
- Need to weigh building out admin interface allow support to manage user accounts vs allowing access to AWS Console.

**Comment by sarahhobson** [comment]
@bryan-robitaille could you confirm what could be done with access to AWS console?

**Comment by anikbrazeau** [comment]
Do we have any documented use cases or scenarios so far that might lead us to need to lock/deactivate accounts, or is identifying those part of the scope of this card?

**Comment by timarney** [comment]
Would be good to consult with the Notify team - not sure if it still exists but but we had "block user" as part of the admin 

https://github.com/cds-snc/notification-admin/pull/279/files#diff-e15cc8ebf52f99875986ca7750e6f632cfa1b5772351e5b0a704faf1b7acccceR37

Which would lock out but not delete an account.  Not sure how often it's been used but handy while investigating a potential threat.

**Comment by bryan-robitaille** [comment]
There are two parts places where a user exists in our product: Cognito, and GCForms.  A user can be disabled or deleted in one but not necessarily the other so we'll need to figure out as a Product where we want to apply those limitations.

For example.
Disabled in GCForms but not Cognito
- A user enters their username and password
- Receives the 2fa code and enters it correctly
- Is immediately redirected to a screen informing them they are persona non grata.

Disabled in Cognito but not necessarily GCForms
 - User enters username and password
 - Immediate receives a message that their account is locked
 

Each scenario has it's pros and cons and level of potential risk it could introduce in the app.

**Comment by srtalbot** [comment]
Would a call with the folks on this card help to walk through the scenarios and trade-offs?

**Comment by samsadasivan** [comment]
@srtalbot I think that is a good idea - I can definitely map out the scenarios from a experience design view, but it would be awesome to know the constraints up front. 

I think we can give @bryan-robitaille some time to do some exploration on the pros and cons of both options, then regroup next week?

**Comment by samsadasivan** [comment]
I can still get started on some of the items on the checklist - not blocked by this

**Comment by anikbrazeau** [comment]
Some early thinking/exploration of the words we might consider using: https://www.figma.com/file/efgWdO13FJfgY20ZKPcLC2/Content-sandbox?type=whiteboard&node-id=0%3A1&t=O1hILS7bsne8vfl6-1

`suspend`
`lock`
`block`
`archive`
`delete`
`revoke`
`deactivate`
`remove`
`disable`

**Comment by samsadasivan** [comment]
Card handed off to @connorscarolyns

**Comment by connorscarolyns** [comment]
User-flows for locking and deactivating users are complete. They can be reviewed in more detail [here](https://www.figma.com/file/Rbs7RPSDUhYeT7OCMF4o3G/Locking-out-and-deactivating-users?type=whiteboard&node-id=0%3A1&t=vdMgBVZECIocEqTV-1). ![image.png](https://images.zenhubusercontent.com/620427f3846ebb620d78e582/418df5c9-10d9-48a8-8715-14dabbfcc7c2)![image.png](https://images.zenhubusercontent.com/620427f3846ebb620d78e582/a149272a-b4d2-448f-9cfb-289332965aa4)![image.png](https://images.zenhubusercontent.com/620427f3846ebb620d78e582/77d49868-2288-40ca-bf73-f09648bb766e)

**Comment by connorscarolyns** [comment]
@bryan-robitaille has reviewed for technical feasibility, @srtalbot & @sarahhobson have been stages for product  and policy review.

**Comment by connorscarolyns** [comment]
A separate ticket was created for deactivation documentation
